### PR TITLE
Issue 414 - Add File Inputs to Jobs Table

### PR DESCRIPTION
**Pull request by steveb-ais**
* Add useful random file input objects to mock jobs API returns
* Adds Truncate pipe for trimming text by character length
* Adds array flatten utility
* Adds input_files key to the Jobs model
* Updates Jobs API results with sorted, unique, and flattened string.

### CODE REVIEW

File: projects/developer/src/app/common/pipes/truncate.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+    name: 'truncate',
+})
+export class TruncatePipe implements PipeTransform {
+    transform(value: string, limit = 25, completeWords = false, ellipsis = '...') {
```
Comment (mheppner) [inline projects/developer/src/app/common/pipes/truncate.pipe.ts]: Can we just use the lodash version?

File: projects/developer/src/app/common/utils/ArrayUtilities.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export function flatDeep(arr, depth = 1) {
+    return depth > 0
```
Comment (mheppner) [inline projects/developer/src/app/common/utils/ArrayUtilities.ts]: Lodash?

File: projects/developer/src/app/processing/jobs/component.html [hunk 3]
```diff
@@ -90,6 +90,13 @@ <h2><i class="fa fa-cube"></i> Jobs <span *ngIf="count">({{ count }})</span></h2
                         </a>
                     </div>
                 </div>
+                <div *ngSwitchCase="'inputs'">
+                    <div *ngIf="rowData.input_files">
+                        <span [tooltipStyleClass]="'max-size'" [pTooltip]="makeTooltip(rowData.input_files)">
```
Comment (mheppner) [inline projects/developer/src/app/processing/jobs/component.html]: Very minor, but can you just leave this without the binding? `<span tooltipStyleClass="max-size" ...`

Comment (steveb-ais) [reply projects/developer/src/app/processing/jobs/component.html]: Done.

File: projects/developer/src/app/common/utils/ArrayUtilities.ts (hunk 2 above)
Comment (steveb-ais) [reply projects/developer/src/app/common/utils/ArrayUtilities.ts]: Replaced with lodash. :-\

File: projects/developer/src/app/common/pipes/truncate.pipe.ts (hunk 1 above)
Comment (steveb-ais) [reply projects/developer/src/app/common/pipes/truncate.pipe.ts]: Replaced truncate pipe logic with lodash.

**Pull request merged**